includes/Mapper/class-contactmapper.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * Contact mapper
+ *
+ * @package OmnisendPaidMerbershipsProPlugin
+ */
+
+declare(strict_types=1);
+
+namespace Omnisend\PaidMembershipsProAddon\Mapper;
+
+use Omnisend\PaidMembershipsProAddon\Actions\OmnisendAddOnAction;
+use Omnisend\SDK\V1\Contact;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class ContactMapper
+ */
+class ContactMapper {
+	private const CUSTOM_PREFIX  = 'paid_memberships_pro';
+	private const CONSENT_PREFIX = 'paid_memberships_pro';
+
+	/**
+	 * Create/update Omnisend contact
+	 *
+	 * @param array  $mapped_fields
+	 *
+	 * @return Contact object
+	 */
+	public function create_contact( array $mapped_fields ): Contact {
+		$contact = new Contact();
+
+		$current_user = wp_get_current_user();
+		$user_email   = $current_user->user_email;
+
+		if ( 0 != $current_user->ID ) {
+			$user_email = $mapped_fields['bemail'];
+		}
+
+		$contact->set_email( $user_email );
+		$contact->set_phone( $mapped_fields['bphone'] );
+		$contact->set_first_name( $mapped_fields['bfirstname'] ?? '' );
+		$contact->set_last_name( $mapped_fields['blastname'] ?? '' );
+		$contact->set_postal_code( $mapped_fields['bzipcode'] ?? '' );
+		$contact->set_address( $mapped_fields['baddress1'] ?? '' );
+		$contact->set_state( $mapped_fields['bstate'] ?? '' );
+		$contact->set_country( $mapped_fields['bcountry'] ?? '' );
+		$contact->set_city( $mapped_fields['bcity'] ?? '' );
+		$contact->set_welcome_email( true );
+
+		if ( ! isset( $options['setting_field'] ) ) {
+			if ( isset( $mapped_fields['bconsentEmail'] ) ) {
+				$contact->set_email_subscriber();
```

Comment:
It's working fine since we updated omnisend SDK plugin by adding set_email_subscriber, set_email_unsubscriber and set_phone_subscriber, set_phone_unsubscriber. 
Setting email_opt_in or phone_opt_in now sets email_status and phone_status. to 'subscribed' so in this case using only set_email_subscriber and set_email_unsubscriber sets these statuses and user never will be nonsubscriber unless we will not set these statuses (but in these forms we want to set them since client can update them everytime form is submited)